store/postgres/src/chain_store.rs

```diff
@@ -206,6 +213,69 @@ impl ChainStore {
             },
         )
     }
+
+    /// Find the first block that is missing from the database needed to
+    /// complete the chain from block `hash` to the block with number
+    /// `first_block`. We return the hash of the missing block as an
+    /// array because the remaining code expects that, but the array will only
+    /// ever have at most one element.
```

Comment:
The original sql I believe could in fact return an array, if you had a state in the DB such as:
```
*missing* <- B2 <- *missing* <- B4
```

Then iirc it would return both B1 and B3.

Reply:
I don't think that can happen since the recursive query will stop as soon as `b.parent_hash is null` - that sets `chain.last` to `true` and stops the loop.